src/problem/generator/linearSystem.ts

```diff
@@ -0,0 +1,164 @@
+import { ProblemGenerator } from "../ProblemGeneratorSpi";
+import { randomInt } from "../util/randomizer";
+import { exclude } from "../util/predicates";
+import { fracTex } from "../util/texGenerator";
+import { Fraction } from "../util/commonDivisor";
+
+type EquationTerm = {
+    a: number;
+    b: number;
+    c: number;
+    d: number;
+    e: number;
+    };
+
+
+type Params = {
+    terms: EquationTerm[];
+    solutions: Fraction[];
+    problem: string | undefined;
+    };
+
+
+const INFINITY1 = "INFINITY1";
+
+
+////////////////////////////////////////////////////
+//   LEVEL 1 (10%)
+//
+//   b1 x + c1 y = e1
+//   b2 x + c2 y = e2
+//
+///////////////////////////////////////////////////
+
+const level1 = (): Params => {
+
+
+    const x = randomInt(-9, 10, exclude(0));
+    const y = randomInt(-9, 10, exclude(0));
+    const b1 = randomInt(-6,7,exclude(0));
+    const b2 = randomInt(-6,7,exclude(0));
+    const c1 = randomInt(-6,7,exclude(0));
+    const c2 = randomInt(-6,7,exclude(0));
+    const e1 = b1 * x + c1 * y;
+    const e2 = b2 * x + c2 * y;
+        
+    const problem =
+         b1 * c2 - b2 * c1 === 0
+         ? INFINITY1
+         : undefined;
+
+
+    return {
+        terms: [{ a: 0, b: b1, c: c1, d: 0, e: e1},
+                { a: 0, b: b2, c: c2, d: 0, e: e2}],
+        solutions: 
+            [{ a: x, b: 1 },{ a: y, b: 1 }],
+        problem:
+            problem,
+    };
+  };
+
+  export const calculateParameter = (level: number): Params => {
+    switch (level) {
+      case 1:
+        return level1();
+  /*    
+      case 2:
+        return level2();
+        
+      case 3:
+          return level3();
+  
+      case 4:
+        return level4();
+      
+      case 5:
+        return level5();
+    
+      case 6:
+        return level6();
+    */       
+      default:
+        return level1();
+    }
+  };
+  
+  export const renderEquation = (params: Params) => {
+    const textEquation =
+    `\\begin{aligned}
+    \\begin{cases}`
+    + params.terms
+    .map((term) => `${term.a}w+${term.b}x+${term.c}y+${term.d}z=${term.e}`)
+    .join("\\\\")
+    + `\\end{cases}
+  \\end{aligned}`;
+
+  
+    return textEquation
+      .replaceAll("+-", "-")
```

Comment:
Es muss da einen besseren Weg geben ;-)
Vielleicht etwas in Richtung mathjs parse & simplify ([see](https://mathjs.org/index.html))